frame/support/test/tests/pallet.rs

```diff
@@ -1590,6 +1590,30 @@ fn metadata() {
 	pretty_assertions::assert_eq!(actual_metadata.pallets, expected_metadata.pallets);
 }
 
+#[test]
+fn metadata_at_version() {
```

Comment:
Can you fetch the current metadata version and not hardcode it to `14` below in the test?

In other words not to break this test on each metadata upgrade...

Reply:
Since the feature-based approach, I've also added a few dummy tests under `metadata-v14` flag.
Hopefully, we won't have to break the metadata that often (since we have tons of tests that depend on v14 already).

I think we could extend the runtime to always implement for testing purposes `fn metadata_v[num]` if the feature is enabled, then we could make sure to not break tests in the future. Let me know what you think, thanks!

Reply:
maybe have a constant somewhere central containing the current version that you can then use in the tests?